app/views/govuk_publishing_components/components/_search.html.erb

```diff
@@ -13,6 +14,13 @@
   data_attributes ||= {}
   data_attributes[:module] = 'gem-track-click'
 
+  label_classes = []
+  if (shared_helper.valid_heading_size?(label_size))
+    label_classes << "govuk-label govuk-label--#{label_size}"
+  else
+    label_classes << "gem-c-search__label"
+  end
```

Comment:
Sorry, don't want to throw a spanner in the works late in the day but have you checked this option when against a dark background? If we discard the `gem-c-search__label` class for the custom label size then the stuff that's already in place doesn't get used, and the label has black text, which for smaller text sizes is a contrast fail.

You might have to remove the size styling from `gem-c-search__label` and put in a default govuk label size, in order to keep that class by default.

Reply:
Thanks for pointing this out @andysellick. We're not of any instance where this happens. If it does crop up we'll fix it of course.

Reply:
My preferred option would be to completely remove the `gem-c-search__label` styles and switch to using the Design System styles - but that'd be a breaking change and not one that I'm keen to embark on at the moment.

This shouldn't be an issue as all of the current uses of this component don't set at `label_size` parameter, so they'll retain the `gem-c-search__label` styles.